### PR TITLE
Fix command error 'area is already set'

### PR DESCRIPTION
**Pull request by haelbichalex**

### CODE REVIEW

**Comment by Ethan3600** [comment]
@haelbichalex Sorry for the inactivity, I'll try to review this PR this week!

File: Console/Command/Runjob.php [hunk 1]
```diff
@@ -63,7 +63,9 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $dateTime = $this->dateTimeFactory->create();
 
         try {
-            $this->state->setAreaCode('adminhtml');
+            if (!$this->state->getAreaCode()) {
+                $this->state->setAreaCode('adminhtml');
```
Comment (Ethan3600) [inline Console/Command/Runjob.php]: Let's use `Area::AREA_ADMINHTML` here too

**Review by Ethan3600 — CHANGES_REQUESTED** [review]
One tiny change and we can get this merged in!

**Comment by haelbichalex** [comment]
Done :white_check_mark:

**Comment by Ethan3600** [comment]
Just want to confirm one thing here: https://github.com/Ethan3600/magento2-CronjobManager/pull/77/files#diff-2e06a05bbdb681d5be7c59da0d9a3ee4R67

`getAreaCode()` [throws an exception](https://github.com/magento/magento2/blob/2.3-develop/lib/internal/Magento/Framework/App/State.php#L151), if the area code is not set.

Is the expectation here to catch that exception? https://github.com/Ethan3600/magento2-CronjobManager/blob/9c490d5db3b162d7e4ec7a01b784b168ec773130/Console/Command/Runjob.php#L79

In the above scenario, the command will always fail. Not good.

Maybe we should remove the setAreaCode() outside the try/catch that it's in and move it it's own try/catch?

like this:

```php
try {
    $this->state->setAreaCode(Area::AREA_ADMINHTML);
} catch (LocalizedException $e) {
    // Area code is already set
}

try {
    // lets create a new cron job and dispatch it
    $jobCode = $input->getArgument(self::INPUT_KEY_JOB_CODE);
    $now = strftime('%Y-%m-%dT%H:%M:%S', $dateTime->gmtTimestamp());
    ..... rest of the code here
```

This way, we're always attempting to set the areaCode, and if it's already set, then the command will still run.

**Comment by Ethan3600** [comment]
@haelbichalex I think it's worth making that change as I mentioned in the previous comment. I believe it's more robust. Let me know what you think!

**Comment by haelbichalex** [comment]
@Ethan3600 Sorry for the delay. As suggested I moved the `setAreaCode()` to its own try-catch block as its independent of the other exception. Did work like expected on my M2 instance 👍

**Pull request merged**